dspace-api/src/main/java/org/dspace/subscriptions/ContentGenerator.java

```
@@ -0,0 +1,94 @@
+import org.dspace.core.Email;
+import org.dspace.core.I18nUtil;
+import org.dspace.discovery.IndexableObject;
+import org.dspace.eperson.EPerson;
+import org.dspace.subscriptions.service.SubscriptionGenerator;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * Implementation class of SubscriptionGenerator
+ * which will handle the logic of sending the emails
+ * in case of content subscriptions
+ */
+@SuppressWarnings("rawtypes")
+public class ContentGenerator implements SubscriptionGenerator<IndexableObject> {
+
+    private final Logger log = LogManager.getLogger(ContentGenerator.class);
+
+    @SuppressWarnings("unchecked")
+    @Resource(name = "entityDissemination")
+    private Map<String, StreamDisseminationCrosswalk> mapEntityDisseminatorProperty = new HashMap();
+
+    @Autowired
+    private ItemService itemService;
+
+    @Override
+    public void notifyForSubscriptions(Context context, EPerson ePerson,
+                                       List<IndexableObject> indexableComm,
+                                       List<IndexableObject> indexableColl,
+                                       List<IndexableObject> indexableItems) {
+        try {
+            if (Objects.nonNull(ePerson)) {
+                Locale supportedLocale = I18nUtil.getEPersonLocale(ePerson);
+                Email email = Email.getEmail(I18nUtil.getEmailFilename(supportedLocale, "subscriptions_content"));
+                email.addRecipient(ePerson.getEmail());
+                email.addArgument(generateHtmlBodyMail(context, indexableComm));
+                email.addArgument(generateHtmlBodyMail(context, indexableColl));
+                email.addArgument(generateHtmlBodyMail(context, indexableItems));
+                email.send();
+            }
+        } catch (Exception e) {
+            // log this email error
+            log.warn("Cannot email user eperson_id:" + ePerson.getID() + " eperson_email:" + ePerson.getEmail());
```

Here we lose all of the information about *how* the email operation failed.  In my case it appears that the cause is that the `subscriptions_content` email template is not provided by this patch and thus is missing.  Please report the Exception's message, and perhaps a stack trace.

Also, we should be using the more efficient facilities of Log4J v2:
`log.warn("Cannot email user eperson_id: {} eperson_email: {}", ePerson::getID, ePerson::getEmail);`